include/rocksdb/filter_policy.h

```diff
@@ -65,6 +65,32 @@ class FilterBitsBuilder {
   // The ownership of actual data is set to buf
   virtual Slice Finish(std::unique_ptr<const char[]>* buf) = 0;
 
+  // Similar to Finish(std::unique_ptr<const char[]>* buf), except that
+  // for a non-null status pointer argument, it should point to
+  // Status::Corruption() when there is any corruption during filter
+  // construction or Status::OK() otherwise.
+  virtual Slice Finish(std::unique_ptr<const char[]>* buf,
+                       Status* /* status */) {
+    return Finish(buf);
+  }
+
+  // Verify the filter returned from calling FilterBitsBuilder::Finish.
+  // The function returns Status::Corruption() if there is any corruption on the
+  // constructed filter or Status::OK() otherwise.
+  //
+  // The verification will only take effects
+  // when BlockBasedTableOptions::detect_filter_construct_corruption = true.
+  // Otherwise it is no-op and always returns Status::OK().
```

Comment:
This should instead say something like "Implementations should normally consult FilterBuildingContext::table_options.detect_filter_construct_corruption to determine whether to perform verification or to skip by returning Status::OK(). The decision is left to the FilterBitsBuilder so that verification prerequisites before PostVerify can be skipped when not configured." Does my wording make sense?